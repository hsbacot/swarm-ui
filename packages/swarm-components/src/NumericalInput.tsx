import * as React from 'react';
import { Icon } from './Icon';

type Value = number | null | undefined;

interface Props {
	/**
	 * Classname of wrapper, specified because it is not applied with other
	 */
	className?: string,
	/**
	 * indicates whether the input is disabled
	 */
	disabled?: boolean,
	/**
	 * indicates whether there is an error on the input
	 */
	error?: React.ReactElement,
	/**
	 * the maximum integer allowed
	 */
	max?: number | undefined,
	/**
	 * the minimum integer allowed
	 */
	min?: number | undefined,
	/**
	 * Required change handler with Value, not event
	 */
	onChange: (value: Value) => void,
	/**
	 * funcitonality invo
	 * ked when the form field is blurred
	 */
	onBlur?: () => void,
	/**
	 * The amount the input will increment or decrement when using keyboard interactions
	 */
	step: number,
	/**
	 * The value of the field
	 */
	value: Value,
};

type State = {
	isFieldFocused: boolean,
	value: Value,
};

export const getStatus = (props: Props): string => {
	return props.disabled ? 'disabled' : props.error ? 'error' : 'default';
};

export const isDefined = (number: Value): boolean => {
	return typeof number !== 'undefined' && number !== null;
};

class NumericalInput extends React.Component<Props, State> {
	fauxInputEl: HTMLInputElement | null | undefined;
	decrementBtnEl: HTMLButtonElement | null | undefined;
	incrementBtnEl: HTMLButtonElement | null | undefined;

	static defaultProps = {
		step: 1,
	};

	state = {
		value: isDefined(this.props.value) ? this.props.value : null,
		isFieldFocused: false,
	};

	// internal value updater when passed a value via props
	static getDerivedStateFromProps(nextProps: Props, prevState: State) {
		const isNewValue = nextProps.onChange && nextProps.value !== prevState.value;

		return {
			value: isNewValue ? nextProps.value : prevState.value,
		};
	}

	_updateValueByStep = (isIncreasing: boolean) => {
		const min: number = isDefined(this.props.min)
			? Number(this.props.min)
			: -Infinity;
		const max: number = isDefined(this.props.max) ? Number(this.props.max) : Infinity;
		const step: number = this.props.step || 1;

		const currentVal: number = this.state.value
			? this.state.value
			: this.props.min
			? this.props.min
			: 0;
		const newValue = isIncreasing ? currentVal + step : currentVal - step;
		const minConstrainedValue = Math.max(newValue, min);

		return Math.min(minConstrainedValue, max);
	};

	// private value updater for keyboard and button interactions
	_updateValue = (value: Value) => {
		this.setState({ value });

		if (this.props.onChange) {
			this.props.onChange(value);
		}
	};

	// this signals blur for the input and the two button controls
	// only registers a blur when all elements are blurred
	onBlur = (e: React.FocusEvent<HTMLInputElement | HTMLButtonElement>) => {
		const formControls = [this.fauxInputEl, this.decrementBtnEl, this.incrementBtnEl];
		if (formControls.every(c => c !== document.activeElement)) {
			this.setState({ isFieldFocused: false });
			if (this.props.onBlur) {
				this.props.onBlur(e);
			}
		}
	};

	onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		const newValue = parseInt(e.target.value, 10);
		if (Number.isNaN(newValue)) {
			return;
		}
		this._updateValue(newValue);
	};

	// this signals focus for the inpurt and the two button controls
	onFocus = (e: React.FocusEvent<HTMLInputElement | HTMLButtonElement>) => {
		this.setState({ isFieldFocused: true });
	};

	onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
		// Disable the 'e' or 'E' values because we don't
		// support scientific notation at the moment
		if (e.key.toLowerCase() === 'e') {
			e.preventDefault();
		}
	};

	// the event for the increment button
	incrementAction = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
		e.preventDefault();
		this._updateValue(this._updateValueByStep(true));
	};

	// the event for the decrement button
	decrementAction = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
		e.preventDefault();
		this._updateValue(this._updateValueByStep(false));
	};

	render() {
		const {
			className = '',
			onChange, // eslint-disable-line no-unused-vars
			onBlur, // eslint-disable-line no-unused-vars
			value, // eslint-disable-line no-unused-vars
			...other
		} = this.props;

		return (
			<div data-swarm-number-input={getStatus(this.props)} className={className}>
				<input
					data-swarm-number-input-field
					type="number"
					pattern="[0-9]*"
					onBlur={this.onBlur}
					onFocus={this.onFocus}
					onChange={this.onChange}
					onKeyDown={this.onKeyDown}
					value={this.state.value === null ? '' : this.state.value}
					{...other}
				/>
				<button
					data-swarm-number-input-decrement
					tabIndex={-1}
					onBlur={this.onBlur}
					onClick={this.decrementAction}
					onFocus={this.onFocus}
					ref={(el: HTMLButtonElement | null) => (this.decrementBtnEl = el)}
				>
					<Icon shape="minus" />
				</button>
				<button
					data-swarm-number-input-increment
					tabIndex={-1}
					onBlur={this.onBlur}
					onClick={this.incrementAction}
					onFocus={this.onFocus}
					ref={(el: HTMLButtonElement | null) => (this.incrementBtnEl = el)}
				>
					<Icon shape="add" />
				</button>
			</div>
		);
	}
}

export { NumericalInput };
