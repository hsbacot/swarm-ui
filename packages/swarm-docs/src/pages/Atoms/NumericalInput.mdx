import { NumericalInput } from '@meetup/swarm-components';
import Example from '../../components/examples/numerical';

# Numerical Input
Fills to the width of its container.
```.jsx
<div style={{ width: 300 }}>
  <NumericalInput value={1} />
</div>
```

## example
<Example />
