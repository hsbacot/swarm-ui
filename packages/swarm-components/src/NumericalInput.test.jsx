import React from 'react';
import { shallow } from 'enzyme';
import { NumericalInput, getStatus } from './NumericalInput';

describe('NumericalInput', () => {
	describe('render', () => {
		const mockOnChange = jest.fn();
		const mockOnBlur = jest.fn();
		const testCases = [
			['Default', <NumericalInput key="0" value={null} onChange={mockOnChange} />],
			[
				'Default with classname',
				<NumericalInput
					key="1"
					value={null}
					onChange={mockOnChange}
					className="foo"
				/>,
			],
			[
				'Default with id',
				<NumericalInput key="2" value={null} onChange={mockOnChange} id="foo" />,
			],
			[
				'Default with step',
				<NumericalInput key="3" value={null} onChange={mockOnChange} step={2} />,
			],
			[
				'Default with max',
				<NumericalInput key="4" value={null} onChange={mockOnChange} max={10} />,
			],
			[
				'Default with min',
				<NumericalInput key="5" value={null} onChange={mockOnChange} min={1} />,
			],
			[
				'Default with onBlur',
				<NumericalInput
					key="6"
					value={null}
					onChange={mockOnChange}
					onBlur={mockOnBlur}
				/>,
			],
			[
				'Disabled',
				<NumericalInput key="7" value={null} onChange={mockOnChange} disabled />,
			],
			[
				'Error',
				<NumericalInput key="8" value={null} onChange={mockOnChange} error />,
			],
		];

		test.each(testCases)('Snapshot: %s', (description, element) => {
			expect(shallow(element)).toMatchSnapshot();
		});
	});

	describe('getDerivedStateFromProps', () => {
		const prevState = {
			value: 4,
		};

		it('should return a new state value when state has changed', () => {
			const nextProps = {
				onChange: jest.fn(),
				value: 5,
			};

			expect(NumericalInput.getDerivedStateFromProps(nextProps, prevState)).toEqual({
				value: nextProps.value,
			});
		});

		it('should return same state value when state has not changed', () => {
			const nextProps = {
				onChange: jest.fn(),
				value: 4,
			};

			expect(NumericalInput.getDerivedStateFromProps(nextProps, prevState)).toEqual({
				value: prevState.value,
			});
		});
	});

	describe('_updateValueByStep', () => {
		describe('with no min, max, or step prop', () => {
			it('should increase value by 1 when increasing', () => {
				const numericalInputComponent = shallow(
					<NumericalInput value={null} onChange={jest.fn()} />
				);
				expect(numericalInputComponent.instance()._updateValueByStep(true)).toEqual(
					1
				);
			});

			it('should decrease value by 1 when decreasing', () => {
				const numericalInputComponent = shallow(
					<NumericalInput value={null} onChange={jest.fn()} />
				);
				expect(numericalInputComponent.instance()._updateValueByStep(false)).toEqual(
					-1
				);
			});
		});

		describe('with step prop only', () => {
			it('should increase value by `step` when increasing', () => {
				const numericalInputComponent = shallow(
					<NumericalInput value={null} onChange={jest.fn()} step={2} />
				);
				expect(numericalInputComponent.instance()._updateValueByStep(true)).toEqual(
					2
				);
			});

			it('should decrease value by `step` when decreasing', () => {
				const numericalInputComponent = shallow(
					<NumericalInput value={null} onChange={jest.fn()} step={2} />
				);
				expect(numericalInputComponent.instance()._updateValueByStep(false)).toEqual(
					-2
				);
			});
		});

		describe('with min prop only', () => {
			it('should decrease by `1` when value stays greater than min', () => {
				const numericalInputComponent = shallow(
					<NumericalInput value={1} onChange={jest.fn()} min={0} />
				);
				expect(numericalInputComponent.instance()._updateValueByStep(false)).toEqual(
					0
				);
			});

			it('should not decrease past `min`', () => {
				const numericalInputComponent = shallow(
					<NumericalInput value={0} onChange={jest.fn()} min={0} />
				);
				expect(numericalInputComponent.instance()._updateValueByStep(false)).toEqual(
					0
				);
			});
		});

		describe('with max prop only', () => {
			it('should increase by `1` when value stays less than max', () => {
				const numericalInputComponent = shallow(
					<NumericalInput value={0} onChange={jest.fn()} max={10} />
				);
				expect(numericalInputComponent.instance()._updateValueByStep(true)).toEqual(
					1
				);
			});

			it('should not increase past `max`', () => {
				const numericalInputComponent = shallow(
					<NumericalInput value={10} onChange={jest.fn()} max={10} />
				);
				expect(numericalInputComponent.instance()._updateValueByStep(true)).toEqual(
					10
				);
			});
		});
	});

	describe('_updateValue', () => {
		it('calls setState', () => {
			let numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={jest.fn()} />
			);

			numericalInputComponent.setState = jest.fn(numericalInputComponent.setState);
			numericalInputComponent.instance()._updateValue(5);

			expect(numericalInputComponent.setState).toHaveBeenCalled();
		});

		it('calls the `onChange` callback prop', () => {
			const mockOnChange = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);

			numericalInputComponent.instance()._updateValue(5);
			expect(mockOnChange).toHaveBeenCalled();
		});
	});

	describe('onBlur', () => {
		it('should call the onBlur prop', () => {
			const mockOnChange = jest.fn();
			const mockOnBlur = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} onBlur={mockOnBlur} />
			);
			const numericalInputInstance = numericalInputComponent.instance();

			global.document = {
				activeElement: numericalInputInstance.fauxInputEl,
			};
			numericalInputInstance.onBlur({});

			expect(mockOnBlur).toHaveBeenCalled();
		});

		it('should state `isFieldFocused` state to false', () => {
			const mockOnChange = jest.fn();
			const mockOnBlur = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} onBlur={mockOnBlur} />
			);
			const numericalInputInstance = numericalInputComponent.instance();
			global.document = {
				activeElement: numericalInputInstance.fauxInputEl,
			};
			numericalInputInstance.onBlur({});

			expect(numericalInputComponent.state('isFieldFocused')).toEqual(false);
		});
	});

	describe('onChange', () => {
		it('should call the onChange prop', () => {
			const mockOnChange = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);
			const numericalInputInstance = numericalInputComponent.instance();

			numericalInputInstance.onChange({
				target: {
					value: '10',
				},
			});

			expect(mockOnChange).toHaveBeenCalled();
		});

		it('should not call onChange when target value is not a number', () => {
			const mockOnChange = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);
			const numericalInputInstance = numericalInputComponent.instance();

			numericalInputInstance.onChange({
				target: {
					value: 'foo',
				},
			});

			expect(mockOnChange).not.toHaveBeenCalled();
		});
	});

	describe('onFocus', () => {
		it('should set `isFieldFocused` state to true', () => {
			const mockOnChange = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);
			const numericalInputInstance = numericalInputComponent.instance();
			numericalInputInstance.onFocus({});

			expect(numericalInputComponent.state('isFieldFocused')).toEqual(true);
		});
	});

	describe('onKeyDown', () => {
		it('should preventDefault on the event if key is an `E`', () => {
			const mockOnChange = jest.fn();
			const mockPreventDefault = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);
			const numericalInputInstance = numericalInputComponent.instance();
			numericalInputInstance.onKeyDown({
				key: 'E',
				preventDefault: mockPreventDefault,
			});

			expect(mockPreventDefault).toHaveBeenCalled();

			mockPreventDefault.mockClear();
			numericalInputInstance.onKeyDown({
				key: 'e',
				preventDefault: mockPreventDefault,
			});

			expect(mockPreventDefault).toHaveBeenCalled();
		});

		it('should not preventDefault is key is not an `E`', () => {
			const mockOnChange = jest.fn();
			const mockPreventDefault = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);
			const numericalInputInstance = numericalInputComponent.instance();
			numericalInputInstance.onKeyDown({
				key: '$',
				preventDefault: mockPreventDefault,
			});

			expect(mockPreventDefault).not.toHaveBeenCalled();
		});
	});

	describe('incrementAction', () => {
		it('should call e.preventDefault', () => {
			const mockOnChange = jest.fn();
			const mockPreventDefault = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);
			const numericalInputInstance = numericalInputComponent.instance();

			numericalInputInstance.incrementAction({ preventDefault: mockPreventDefault });

			expect(mockPreventDefault).toHaveBeenCalled();
		});

		it('should call this._updateValueByStep', () => {
			const mockOnChange = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);
			const numericalInputInstance = numericalInputComponent.instance();
			numericalInputInstance._updateValueByStep = jest.fn();
			numericalInputInstance.incrementAction({ preventDefault: jest.fn() });

			expect(numericalInputInstance._updateValueByStep).toHaveBeenCalledWith(true);
		});

		it('should call this._updateValue', () => {
			const mockOnChange = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);
			const numericalInputInstance = numericalInputComponent.instance();
			numericalInputInstance._updateValue = jest.fn();

			const stepValue = numericalInputInstance._updateValueByStep(true);
			numericalInputInstance.incrementAction({ preventDefault: jest.fn() });

			expect(numericalInputInstance._updateValue).toHaveBeenCalledWith(stepValue);
		});
	});

	describe('decrementAction', () => {
		it('should call e.preventDefault', () => {
			const mockOnChange = jest.fn();
			const mockPreventDefault = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);
			const numericalInputInstance = numericalInputComponent.instance();

			numericalInputInstance.decrementAction({ preventDefault: mockPreventDefault });

			expect(mockPreventDefault).toHaveBeenCalled();
		});

		it('should call this._updateValueByStep', () => {
			const mockOnChange = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);
			const numericalInputInstance = numericalInputComponent.instance();
			numericalInputInstance._updateValueByStep = jest.fn();
			numericalInputInstance.decrementAction({ preventDefault: jest.fn() });

			expect(numericalInputInstance._updateValueByStep).toHaveBeenCalledWith(false);
		});

		it('should call this._updateValue', () => {
			const mockOnChange = jest.fn();
			const numericalInputComponent = shallow(
				<NumericalInput value={0} onChange={mockOnChange} />
			);
			const numericalInputInstance = numericalInputComponent.instance();
			numericalInputInstance._updateValue = jest.fn();

			const stepValue = numericalInputInstance._updateValueByStep(false);
			numericalInputInstance.decrementAction({ preventDefault: jest.fn() });

			expect(numericalInputInstance._updateValue).toHaveBeenCalledWith(stepValue);
		});
	});
});

describe('getStatus', () => {
	it('returns disabled when props contain disabled=true', () => {
		expect(getStatus({ disabled: true })).toEqual('disabled');
		expect(getStatus({ disabled: true, error: true })).toEqual('disabled');
	});

	it('returns error when props contain error=true', () => {
		expect(getStatus({ error: true })).toEqual('error');
	});

	it('returns default when props do not contain disabled or error', () => {
		expect(getStatus({})).toEqual('default');
	});
});
