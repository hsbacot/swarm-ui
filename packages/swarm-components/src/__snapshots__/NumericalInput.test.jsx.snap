// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericalInput render Snapshot: Default 1`] = `
<div
  className=""
  data-swarm-number-input="default"
>
  <input
    data-swarm-number-input-field={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="[0-9]*"
    step={1}
    type="number"
    value=""
  />
  <button
    data-swarm-number-input-decrement={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="minus"
    />
  </button>
  <button
    data-swarm-number-input-increment={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="add"
    />
  </button>
</div>
`;

exports[`NumericalInput render Snapshot: Default with classname 1`] = `
<div
  className="foo"
  data-swarm-number-input="default"
>
  <input
    data-swarm-number-input-field={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="[0-9]*"
    step={1}
    type="number"
    value=""
  />
  <button
    data-swarm-number-input-decrement={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="minus"
    />
  </button>
  <button
    data-swarm-number-input-increment={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="add"
    />
  </button>
</div>
`;

exports[`NumericalInput render Snapshot: Default with id 1`] = `
<div
  className=""
  data-swarm-number-input="default"
>
  <input
    data-swarm-number-input-field={true}
    id="foo"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="[0-9]*"
    step={1}
    type="number"
    value=""
  />
  <button
    data-swarm-number-input-decrement={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="minus"
    />
  </button>
  <button
    data-swarm-number-input-increment={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="add"
    />
  </button>
</div>
`;

exports[`NumericalInput render Snapshot: Default with max 1`] = `
<div
  className=""
  data-swarm-number-input="default"
>
  <input
    data-swarm-number-input-field={true}
    max={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="[0-9]*"
    step={1}
    type="number"
    value=""
  />
  <button
    data-swarm-number-input-decrement={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="minus"
    />
  </button>
  <button
    data-swarm-number-input-increment={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="add"
    />
  </button>
</div>
`;

exports[`NumericalInput render Snapshot: Default with min 1`] = `
<div
  className=""
  data-swarm-number-input="default"
>
  <input
    data-swarm-number-input-field={true}
    min={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="[0-9]*"
    step={1}
    type="number"
    value=""
  />
  <button
    data-swarm-number-input-decrement={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="minus"
    />
  </button>
  <button
    data-swarm-number-input-increment={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="add"
    />
  </button>
</div>
`;

exports[`NumericalInput render Snapshot: Default with onBlur 1`] = `
<div
  className=""
  data-swarm-number-input="default"
>
  <input
    data-swarm-number-input-field={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="[0-9]*"
    step={1}
    type="number"
    value=""
  />
  <button
    data-swarm-number-input-decrement={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="minus"
    />
  </button>
  <button
    data-swarm-number-input-increment={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="add"
    />
  </button>
</div>
`;

exports[`NumericalInput render Snapshot: Default with step 1`] = `
<div
  className=""
  data-swarm-number-input="default"
>
  <input
    data-swarm-number-input-field={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="[0-9]*"
    step={2}
    type="number"
    value=""
  />
  <button
    data-swarm-number-input-decrement={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="minus"
    />
  </button>
  <button
    data-swarm-number-input-increment={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="add"
    />
  </button>
</div>
`;

exports[`NumericalInput render Snapshot: Disabled 1`] = `
<div
  className=""
  data-swarm-number-input="disabled"
>
  <input
    data-swarm-number-input-field={true}
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="[0-9]*"
    step={1}
    type="number"
    value=""
  />
  <button
    data-swarm-number-input-decrement={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="minus"
    />
  </button>
  <button
    data-swarm-number-input-increment={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="add"
    />
  </button>
</div>
`;

exports[`NumericalInput render Snapshot: Error 1`] = `
<div
  className=""
  data-swarm-number-input="error"
>
  <input
    data-swarm-number-input-field={true}
    error={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    pattern="[0-9]*"
    step={1}
    type="number"
    value=""
  />
  <button
    data-swarm-number-input-decrement={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="minus"
    />
  </button>
  <button
    data-swarm-number-input-increment={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <Icon
      shape="add"
    />
  </button>
</div>
`;
